src/measure_theory/l2_space.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2021 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import analysis.normed_space.inner_product
+import measure_theory.set_integral
+
+/-! # `L^2` space
+
+-/
+
+noncomputable theory
+open topological_space measure_theory measure_theory.Lp
+open_locale nnreal ennreal
+
+namespace measure_theory
+
+variables {α E F G 𝕜 : Type*} [is_R_or_C 𝕜] [measurable_space α] {μ : measure α} {p : ℝ≥0∞}
+  [measurable_space E] [inner_product_space 𝕜 E] [borel_space E] [second_countable_topology E]
+  [normed_group F] [measurable_space F] [borel_space F] [second_countable_topology F]
+  [normed_group G]
+
+lemma two_mul_le_add_sq (a b : ℝ) : 2 * a * b ≤ a ^ 2 + b ^ 2 :=
+begin
+  suffices h_nonneg : 0 ≤ a ^ 2 + b ^ 2 - 2 * a * b, by rwa sub_nonneg at h_nonneg,
+  calc 0 ≤ (a - b) ^ 2               : pow_two_nonneg _
+     ... = a ^ 2 + b ^ 2 - 2 * a * b : by ring,
+end
+
+lemma snorm_rpow_two_norm_lt_top (f : Lp F 2 μ) :
+  snorm (λ (x : α), ∥f x∥ ^ (2 : ℝ)) 1 μ < ∞ :=
+begin
+  have h_two : ennreal.of_real (2 : ℝ) = 2, by simp [zero_le_one],
+  rw [snorm_norm_rpow f zero_lt_two, one_mul, h_two],
+  exact ennreal.rpow_lt_top_of_nonneg zero_le_two (Lp.snorm_ne_top f),
+end
+
+lemma snorm_inner_lt_top (f g : Lp E 2 μ) : snorm (λ (x : α), (inner (f x) (g x) : 𝕜)) 1 μ < ∞ :=
```

Comment:
Can you perhaps use 
```lean
local notation `⟪`x`, `y`⟫` := @inner 𝕜 E _ x y
```
in this file, for clarity?  (Not for the eventual inner product you construct on `Lp E 2 μ`, but on `E` itself.)